icicle/tests/test_hash_api.cpp

```diff
@@ -149,22 +148,25 @@ TEST_F(HashApiTest, MerkleTree)
   auto layer2_hash = HashSumBackend::create(20, 8); // input 20 bytes, output 8 bytes         20B  ->  8B    output
   auto leaf_element_size = 4;
   auto merkle_tree =
-    MerkleTree::create({layer0_hash, layer1_hash, layer2_hash}, leaf_element_size, 2 /*min level to store*/);
+    MerkleTree::create({layer0_hash, layer1_hash, layer2_hash}, leaf_element_size, 1 /*min level to store*/);
 
   // build tree
-  ICICLE_CHECK(merkle_tree.build(leaves, input_size, config));
+  ICICLE_CHECK(merkle_tree.build(leaves, nof_leaves, config));
```

Comment:
we can time this. Interesting to see it compared to CUDA

Reply:
done